Extensions/render/cl-vectors/vectors.lisp

```diff
@@ -103,45 +97,38 @@
   (aa-update-state state paths transformation)
   (aa-cells-sweep/rectangle image design state clip-region))
 
-(defun aa-fill-alpha-paths (image design paths state transformation clip-region)
-  (vectors::state-reset state)
-  (dolist (path paths)
-    (setf (paths::path-type path) :closed-polyline))
-  (aa-update-state state paths transformation)
-  (aa-cells-alpha-sweep/rectangle image design state clip-region))
-
-
-(defun %aa-scanline-sweep (scanline function &key start end)
+(defun %aa-scanline-sweep (scanline function start end)
   "Call FUNCTION for each pixel on the polygon covered by
 SCANLINE. The pixels are scanned in increasing X. The sweep can
 be limited to a range by START (included) or/and END (excluded)."
   (declare (optimize speed (debug 0) (safety 0) (space 2))
-           (type (function (fixnum fixnum fixnum) *) function))
-  (let ((x-min (aa::cell-x (car scanline)))
-        (x-max (aa::cell-x (car scanline)))
-        (cover 0)
-        (y (aa::scanline-y scanline))
-        (cells scanline)
-        (last-x nil))
-    (when start
-      ;; skip initial cells that are before START
-      (loop while (and cells (< (aa::cell-x (car cells)) start))
-         do (incf cover (aa::cell-cover (car cells)))
-         (setf last-x (aa::cell-x (car cells))
-               cells (cdr cells))))
+           (type (function (fixnum fixnum fixnum) *) function)
+           (type fixnum start end))
+  (let* ((x-min (max start (aa::cell-x (first scanline))))
+         (x-max x-min)
+         (cover 0)
+         (y (aa::scanline-y scanline))
+         (cells scanline)
+         (last-x nil))
+    (declare (type fixnum x-min x-max)
+             (type (or null fixnum) last-x))
+    ;; Skip initial cells that are before START.
+    (loop while (and cells (< (aa::cell-x (car cells)) start))
+          do (incf cover (aa::cell-cover (car cells)))
+             (setf last-x (aa::cell-x (car cells))
+                   cells (cdr cells)))
     (when cells
       (dolist (cell cells)
```

Comment:
Do we need this WHEN here?

Reply:
No. Probably some kind of leftover. Changed.